erpnext/hr/doctype/expense_claim/expense_claim.js

```diff
@@ -294,10 +304,41 @@ frappe.ui.form.on("Expense Claim Detail", {
 	}
 });
 
+frappe.ui.form.on("Expense Claim Advance", {
+	employee_advance: function(frm, cdt, cdn) {
+		var child = locals[cdt][cdn];
+		if(!frm.doc.employee){
+			frappe.msgprint('Select an employee to get the employee advance.');
+			frm.get_field("advances").grid.grid_rows[child.idx-1].remove();
```

Comment:
Instead of removing specific row, make the entire table blank `frm.doc.advances= []`